### PR TITLE
Extensive formulation

### PR DESCRIPTION
**Pull request by Henri-Gerard**
Implement the extensive formulation for "damsvalley" problem

### CODE REVIEW

File: examples/extensiveFormulation.jl [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    @addConstraint(mod, x[1,2] == X_init[2])
+    
+    
+    #Define the objective of the function
+    @setObjective(mod, Min, sum{ sum{proba[t][N_SCENARS*(n-1)+k]*c[t,N_SCENARS*(n-1)+k],k = 1:N_SCENARS} , t = 1:T, n=1:div(N[t+1],N_SCENARS)})
+    
+    
+    status = solve(mod)
+    
+    solved = (string(status) == "Optimal")
```
Comment (frapac) [inline examples/extensiveFormulation.jl]: This is a bit hacky (my fault): now, I prefer to use: 

status == :Optimal

File: examples/extensiveFormulation.jl [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    end
+    
+    
+    #Instantiate the problem creating dynamic constraint at each node
+    for t = 1 : (T)
+        println("\n\nt=",t)
+        for n = 1 : N[t]
+            for xi = 1 : laws[t].supportSize
+                m = (n-1)*laws[t].supportSize+xi
+                @addConstraint(mod, [x[t+1,2*(m-1)+1];x[t+1,2*m]] .== model.dynamics(t,[x[t,2*(n-1)+1];x[t,2*n]], [u[t,2*(m-1)+1];u[t,2*m]], laws[t].support[xi]))
```
Comment (frapac) [inline examples/extensiveFormulation.jl]: Could you split this expression in different lines?
@addConstraint(exp1,
                           exp2,
                           ...)

File: src/extensiveFormulation.jl [hunk 3]
```diff
@@ -0,0 +1,95 @@
+#  Copyright 2015, Vincent Leclere, Francois Pacaud and Henri Gerard
```
Comment (frapac) [inline src/extensiveFormulation.jl]: You have got two extensiveFormulation.jl

File: examples/extensiveFormulation.jl [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+#############################################################################
+#  Define the extensive formulation to check the results of small problems
+#############################################################################
+
+
+function extensive_formulation(model, 
+                               param)
+
+
+    const N_SCENARS = 10
```
Comment (frapac) [inline examples/extensiveFormulation.jl]: You don't have to use const inside a function.

**Comment by frapac** [comment]
Good job :)

This is nice to see that we have the same results between SDDP and the extensive formulation !

**Comment by leclere** [comment]
Ok, this is a nice result (I haven't check that it gives the same numerical results but I am relying on you). 

However the "extensive-formulation" part is not finished : the code was done on the example, but we need to have a generic one. Something that solve a SPLinearCostLinearDynamic model by constructing a big linear problem.

For this PR just integrate Francois's remarks and we can merge.

File: examples/extensiveFormulation.jl [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+#############################################################################
+#  Define the extensive formulation to check the results of small problems
+#############################################################################
+
+
+function extensive_formulation(model, 
+                               param)
+
+
+    N_SCENARS = 10
```
Comment (leclere) [inline examples/extensiveFormulation.jl]: Still too much hard coded

File: examples/extensiveFormulation.jl [hunk 6]
```diff
@@ -0,0 +1,102 @@
+#  Copyright 2015, Vincent Leclere, Francois Pacaud and Henri Gerard
+#  This Source Code Form is subject to the terms of the Mozilla Public
+#  License, v. 2.0. If a copy of the MPL was not distributed with this
+#  file, You can obtain one at http://mozilla.org/MPL/2.0/.
+#############################################################################
+#  Define the extensive formulation to check the results of small problems
+#############################################################################
+
+
+function extensive_formulation(model,
```
Comment (leclere) [inline examples/extensiveFormulation.jl:10]: precise the type of model and param

File: examples/extensiveFormulation.jl [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+    #Define the objective of the function
+    @setObjective(mod, Min, sum{ sum{proba[t][N_SCENARS*(n-1)+k]*c[t,N_SCENARS*(n-1)+k],k = 1:N_SCENARS} , t = 1:T, n=1:div(N[t+1],N_SCENARS)})
+    
+    
+    
+    status = solve(mod)
+    
+    solved = (status == :Optimal)
+    
+    if solved
```
Comment (leclere) [inline examples/extensiveFormulation.jl:97]: What happen if it is not solved ?

**Pull request merged**